lib/wallet/src/Cardano/Wallet/DB/Store/Meta/Layer.hs

```diff
@@ -0,0 +1,132 @@
+import qualified Data.Set as Set
+
+
+{-----------------------------------------------------------------------------
+    DB for 'TxMeta'
+------------------------------------------------------------------------------}
+
+-- | A query to the 'TxMeta' database.
+data QueryTxMeta b where
+    GetAll
+        :: Range SlotNo
+        -> Maybe Natural
+        -> SortOrder
+        -> QueryTxMeta [DB.TxMeta]
+    GetOne :: TxId -> QueryTxMeta (Maybe DB.TxMeta)
+
+filterMetas :: Range SlotNo -> [Filter TxMeta]
+filterMetas (Range Nothing Nothing) = []
+filterMetas (Range (Just low) Nothing) = [TxMetaSlot >=. low]
+filterMetas (Range  Nothing (Just high)) = [TxMetaSlot <=. high]
+filterMetas (Range (Just low) (Just high))
+    = [TxMetaSlot >=. low, TxMetaSlot <=. high]
+
+limitMetas :: Maybe Natural -> [SelectOpt record]
+limitMetas Nothing = []
+limitMetas (Just l) = [LimitTo $ fromIntegral l]
+
+orderMetas :: SortOrder -> [SelectOpt TxMeta]
+orderMetas Ascending = [Asc TxMetaSlot, Asc TxMetaTxId]
+orderMetas Descending = [Desc TxMetaSlot, Asc TxMetaTxId]
+
+-- | A 'QueryStore' for 'TxMeta'.
+mkQueryStoreTxMeta :: QueryStore (SqlPersistT IO) QueryTxMeta DeltaTxMetaHistory
+mkQueryStoreTxMeta =
+    QueryStore
+        { queryS = \case
+            GetAll range limit order ->
+                fmap entityVal <$>
+                    selectList @DB.TxMeta
+                        (filterMetas range)
+                        (limitMetas limit <> orderMetas order)
+            GetOne txId ->
+                fmap entityVal <$>
+                    selectFirst @_ @_ @DB.TxMeta
+                        [TxMetaTxId ==. txId]
+                        []
+        , store = mkStoreMetaTransactions
+        }
+
+-- | Get all 'TxId's after a given 'SlotNo'.
+getMetasAfterSlot
+    :: QueryStore (SqlPersistT IO) QueryTxMeta DeltaTxMetaHistory
+    -> SlotNo
+    -> SqlPersistT IO (Set TxId)
+getMetasAfterSlot q s =
+    fmap (foldMap (Set.singleton . txMetaTxId))
+        $ queryS q
+        $ GetAll
+            (Range (Just s) Nothing)
+            Nothing
+            Ascending
+
+instance Query QueryTxMeta where
+    type World QueryTxMeta = TxMetaHistory
+    query :: QueryTxMeta b -> World QueryTxMeta -> b
+    query q (TxMetaHistory allTransactions) = case q of
+      GetAll range mlimit order ->
+        let whichMeta DB.TxMeta{..} = and $ catMaybes
+                [ (txMetaSlot >=) <$> W.inclusiveLowerBound range
+                , (txMetaSlot <=) <$> W.inclusiveUpperBound range
+                ]
+            reorder = case order of
+                Ascending -> sortOn ((,) <$> txMetaSlot <*> txMetaTxId)
+                Descending -> sortOn ((,) <$> Down . txMetaSlot <*> txMetaTxId)
+        in maybe id (take . fromIntegral) mlimit
+                    $ reorder
+                    $ filter whichMeta
+                    $ toList allTransactions
+      GetOne ti -> find ((== ti) . txMetaTxId) allTransactions
```

Comment:
```suggestion
      GetOne ti -> Map.lookup ti allTransactions
```

Just in case we do want to use the in-memory version and then we starting wondering why the query is slow. 😅